src/Services/firebase.js

```diff
@@ -19,66 +21,77 @@ export const authenticateAnonymously = () => {
-        .doc(groceryListId)
-        .get();
+export const getEventById = (id) => {
+    return new Promise((resolve, reject) => {
+        db.collection(process.env.REACT_APP_COLLECTION_KEY).doc(id).get()
+            .then(doc => {
+                resolve(doc.data());
+            })
+            .catch(err => resolve(null));
+    })
 };
 
-export const getGroceryListItems = groceryListId => {
-    return db.collection('groceryLists')
-        .doc(groceryListId)
-        .collection('items')
-        .get();
-}
-
-export const streamGroceryListItems = (groceryListId, observer) => {
-    return db.collection('groceryLists')
-        .doc(groceryListId)
-        .collection('items')
-        .orderBy('created')
-        .onSnapshot(observer);
+const eventDateComparator = (a, b) => {
+    return Date.parse(a['Event End Date']) <  Date.parse(b['Event End Date']) ? -1 : 1
 };
 
-export const addUserToGroceryList = (userName, groceryListId, userId) => {
-    return db.collection('groceryLists')
-        .doc(groceryListId)
-        .update({
-            users: firebase.firestore.FieldValue.arrayUnion({
-                userId: userId,
-                name: userName
-            })
-        });
+const cfpDateComparator = (a, b) => {
+    return Date.parse(a['Call For Proposals End Date']) <  Date.parse(b['Call For Proposals End Date']) ? -1 : 1
 };
 
-export const addGroceryListItem = (item, groceryListId, userId) => {
-    return getGroceryListItems(groceryListId)
-        .then(querySnapshot => querySnapshot.docs)
-        .then(groceryListItems => groceryListItems.find(groceryListItem => groceryListItem.data().name.toLowerCase() === item.toLowerCase()))
-        .then(matchingItem => {
-            if (!matchingItem) {
-                return db.collection('groceryLists')
-                    .doc(groceryListId)
-                    .collection('items')
-                    .add({
-                        name: item,
-                        created: firebase.firestore.FieldValue.serverTimestamp(),
-                        createdBy: userId
-                    });
+export const getOrderedEventsList = orderBy => {
+    // let orderBy = {
+    //     sortBy: 'Event End Date',
+    //     sortDirection: 'asc',
+    //     filters: [
+    //         'ShowPastEvents',
+    //         'RecentlyEndedEvents',
+    //         'CallForProposalsOpen',
+    //     ],
+    // };
+    return new Promise((resolve, reject) => {
+        let query = db.collection(process.env.REACT_APP_COLLECTION_KEY);
+        if(!orderBy.filters.includes('ShowPastEvents')){
+            let curr_date = new Date();
+            query = query.where('Event End Date', '>=', curr_date.toISOString());
+        }
+        if(orderBy.filters.includes('RecentlyEndedEvents')){
+            let curr_date = new Date();
+            query = query.where('Event End Date', '<', curr_date.toISOString());
+            curr_date.setMonth(curr_date.getMonth() - 1);
```

Comment:
Use camelCasing

Reply:
Changed every variable to use camelCasing.